### PR TITLE
androidndkPkgs: Fix unsupported hardening flags & rename `androidndkPkgs_23b`

### PR DESCRIPTION
**Pull request by Artturin**
Got this error
`clang-17: error: unsupported option '-fzero-call-used-regs=used-gpr' for target 'armv7-unknown-linux-android'`

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/androidndk-pkgs/androidndk-pkgs.nix [hunk 1]
```diff
@@ -63,6 +63,7 @@ rec {
     passthru = {
       inherit targetPrefix;
       isClang = true; # clang based cc, but bintools ld
+      hardeningUnsupportedFlags = llvmPackages.clang.cc.hardeningUnsupportedFlagsByTargetPlatform;
```
Comment (JohnRTitor) [inline pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:66]: Why not use something like https://github.com/NixOS/nixpkgs/pull/326166/commits/6cef413dfc05221bbae13befbc67f4a1d6ab00b2

Comment (robbins) [reply pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:66]: Wouldn't that mean we'd have to potentially update it later to disable other flags, whereas with the current code we can just go ahead and get rid of all the unsupported ones for the given arch at once?

Comment (Artturin) [inline pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:66]:
```suggestion
      Inherit (llvmPackages.clang.cc) hardeningUnsupportedFlagsByTargetPlatform;
```

Comment (Artturin) [reply pkgs/development/androidndk-pkgs/androidndk-pkgs.nix]: Yeah

File: pkgs/development/androidndk-pkgs/default.nix [hunk 2]
```diff
@@ -22,6 +22,10 @@ let
       inherit (pkgs)
         stdenv
         runCommand wrapBintoolsWith wrapCCWith;
+
+      # For hardeningUnsupportedFlagsByTargetPlatform
+      llvmPackages = pkgs."llvmPackages_${llvmVersion}";
```
Comment (Pandapip1) [inline pkgs/development/androidndk-pkgs/default.nix:27]: Wouldn't it make more sense to directly pass `llvmPackages` as the argument to `makeNdkPkgs`?

Comment (RossComputerGuy) [reply pkgs/development/androidndk-pkgs/default.nix:27]: Yes, I think that's a better idea

Comment (Artturin) [reply pkgs/development/androidndk-pkgs/default.nix]: Done

**Review by Pandapip1 — APPROVED** [review]
I mean, diff LGTM, but I don't have any way to test this.

**Comment by Artturin** [comment]
> I mean, diff LGTM, but I don't have any way to test this.

This command fails before this PR and works after this PR
`nix build ".#pkgsCross.armv7a-android-prebuilt.hello"`

**Comment by RossComputerGuy** [comment]
Thank you Artturin for the PR, I'll take a look at this tonight. Diff looks good but I'll try myself and leave a nixpkgs-review.

**Comment by RossComputerGuy** [comment]
Unfortunately, I cannot try this PR due to this error:
```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'hello-armv7a-unknown-linux-androideabi-2.12.1'
         whose name attribute is located at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/stdenv/generic/make-derivation.nix:334:7

       … while evaluating attribute 'stdenv' of derivation 'hello-armv7a-unknown-linux-androideabi-2.12.1'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/stdenv/generic/make-derivation.nix:363:14:

          362|       args = attrs.args or ["-e" (attrs.builder or ./default-builder.sh)];
          363|       inherit stdenv;
             |              ^
          364|

       … while evaluating derivation 'stdenv-linux'
         whose name attribute is located at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/stdenv/generic/default.nix:93:14

       … while evaluating attribute 'defaultNativeBuildInputs' of derivation 'stdenv-linux'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/stdenv/generic/default.nix:126:14:

          125|
          126|       inherit initialPath shell
             |              ^
          127|         defaultNativeBuildInputs defaultBuildInputs;

       … from call site

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:125:11:

          124|
          125|   clang = wrapCCWith {
             |           ^
          126|     cc = binaries // {

       … while calling 'wrapCCWith'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/top-level/all-packages.nix:16710:5:

        16709|   wrapCCWith =
        16710|     { cc
             |     ^
        16711|     , # This should be the only bintools runtime dep with this sort of logic. The

       … from call site

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/top-level/all-packages.nix:16729:7:

        16728|     } @ extraArgs:
        16729|       callPackage ../build-support/cc-wrapper (let self = {
             |       ^
        16730|     nativeTools = stdenv.targetPlatform == stdenv.hostPlatform && stdenv.cc.nativeTools or false;

       … while calling 'callPackageWith'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/lib/customisation.nix:212:35:

          211|   */
          212|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          213|     let

       … from call site

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/lib/trivial.nix:955:7:

          954|     { # TODO: Should we add call-time "type" checking like built in?
          955|       __functor = self: f;
             |       ^
          956|       __functionArgs = args;

       … while calling anonymous lambda

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/lib/customisation.nix:136:17:

          135|     in
          136|     mirrorArgs (origArgs:
             |                 ^
          137|     let

       … from call site

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/lib/customisation.nix:138:16:

          137|     let
          138|       result = f origArgs;
             |                ^
          139|

       … while calling anonymous lambda

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/build-support/cc-wrapper/default.nix:8:1:

            7|
            8| { name ? ""
             | ^
            9| , lib

       … while evaluating derivation 'bionic-prebuilt'
         whose name attribute is located at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/stdenv/generic/make-derivation.nix:334:7

       … while evaluating attribute 'buildCommand' of derivation 'bionic-prebuilt'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/build-support/trivial-builders/default.nix:68:16:

           67|         enableParallelBuilding = true;
           68|         inherit buildCommand name;
             |                ^
           69|         passAsFile = [ "buildCommand" ]

       … from call site

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:44:15:

           43|
           44|   buildInfo = ndkBuildInfoFun stdenv.buildPlatform;
             |               ^
           45|   targetInfo = ndkTargetInfoFun stdenv.targetPlatform;

       … while calling 'ndkBuildInfoFun'

         at /nix/store/dplws54x3jakgi5wvwx9dpdnbsinf8aa-source/pkgs/development/androidndk-pkgs/androidndk-pkgs.nix:14:21:

           13|   #
           14|   ndkBuildInfoFun = { config, ... }: {
             |                     ^
           15|     x86_64-apple-darwin = {

       error: Android NDK doesn't support building on aarch64-unknown-linux-gnu, as far as we know
```

**Comment by Artturin** [comment]
@ofborg build pkgsCross.armv7a-android-prebuilt.hello

@RossComputerGuy android ndk doesn't support aarch64-linux https://developer.android.com/ndk/downloads

**Review by JohnRTitor — APPROVED** [review]
LGTM, builds on `x64-linux`.

```
❯  nix build "github:nixos/nixpkgs/refs/pull/328791/merge#pkgsCross.armv7a-android-prebuilt.hello" --print-out-paths
/nix/store/rkdjw7jqzh7fi4iabnap1d4j4354zss0-hello-armv7a-unknown-linux-androideabi-2.12.1
```

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 328791` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Pull request merged**